sycl/include/sycl/types.hpp

```diff
@@ -2208,18 +2238,26 @@ struct VecStorage<bool, N, typename std::enable_if_t<isValidVectorSize(N)>> {
       typename VecStorageImpl<select_apply_cl_t<bool, std::int8_t, std::int16_t,
                                                 std::int32_t, std::int64_t>,
                               N>::DataType;
+  using VectorDataType =
+      typename VecStorageImpl<select_apply_cl_t<bool, std::int8_t, std::int16_t,
+                                                std::int32_t, std::int64_t>,
+                              N>::VectorDataType;
 };
 // Single element signed integers
 template <typename T>
 struct VecStorage<T, 1, typename std::enable_if_t<is_sigeninteger<T>::value>> {
   using DataType = select_apply_cl_t<T, std::int8_t, std::int16_t, std::int32_t,
                                      std::int64_t>;
+  using VectorDataType = select_apply_cl_t<T, std::int8_t, std::int16_t,
+                                           std::int32_t, std::int64_t>;
```

Comment:
```suggestion
  using VectorDataType = DataType;
```

?

Reply:
done, and got the others as well. thanks.